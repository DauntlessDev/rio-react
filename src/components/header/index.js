import React, { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
    Background, Container, Feature, Text, Link as TextLink, ButtonLink, Logo, Darken, Inner, Group, FeatureCallOut, FeatureCallOutBottom, Dropdown,
    SearchIcon,
    SearchInput, Search, SubText,
    Picture, Profile, PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
    return bg ? (
        <Background {...restProps} data-testid="header-bg">
            <Darken>
                {children}
            </Darken>
        </Background>
    ) : <>{children}</>
}


Header.Profile = function HeaderProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>;
};


Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}
Header.FeatureCallOutBottom = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOutBottom {...restProps}>{children}</FeatureCallOutBottom>
}

Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Header.SubText = function HeaderText({ children, ...restProps }) {
    return <SubText {...restProps}>{children}</SubText>
}


Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}


Header.Inner = function HeaderInner({ children, ...restProps }) {
    return <Inner {...restProps}>{children}</Inner>
}



Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>
}



Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <TextLink {...restProps}>{children}</TextLink>
}



Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to}><Logo {...restProps}></Logo></ReactRouterLink>
    )
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [searchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps}>
            <SearchIcon onClick={() => setSearchActive((searchActive) => !searchActive)} data-testid="search-click">
                <img src="/images/icons/search.png" alt="Search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={searchActive}
                data-testid="search-input"
            />
        </Search>
    );
};


Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
    return <PlayButton {...restProps}>{children}</PlayButton>;
};

