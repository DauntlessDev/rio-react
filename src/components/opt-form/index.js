import React from 'react'
import { Container, Input, Button, Text } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
    return (
        <Container{...restProps}>
            {children}
        </Container>
    )
}

OptForm.Input = function OutFormInput({ ...restProps }) {
    return <Input {...restProps} />
}

OptForm.Button = function OutFormButton({ children, ...restProps }) {
    return (
        <Button {...restProps}>
            {children}
            <img src="/images/icons/chevron-right.png" alt="Try Now" />
        </Button>
    )
}

OptForm.Text = function OutFormText({ children, ...restProps }) {
    return (
        <Text {...restProps}>
            {children}
        </Text>
    )
}



OptForm.Break = function OutFormBreak({ ...restProps }) {
    return (
        <Text {...restProps} />
    )
}

