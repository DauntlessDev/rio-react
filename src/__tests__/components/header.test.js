import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Header } from '../../components';

jest.mock('react-router-dom');

describe('<Header />', () => {
    it('renders the basic <Header /> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <Header>
                <Header.Frame>
                    <Header.Logo src="/logo.svg" alt="Netflix" />
                    <Header.TextLink active="true">Hello I am a link!</Header.TextLink>
                </Header.Frame>
                <Header.Inner>
                    Inner
                </Header.Inner>
            </Header>
        );

        expect(getByText('Hello I am a link!')).toBeTruthy();
        expect(getByText('Inner')).toBeTruthy();
        expect(getByTestId('header-bg')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the basic <Header /> without a background', () => {
        const { container, getByText } = render(
            <Header bg={false}>
                <Header.Frame>
                    <Header.Logo src="/logo.svg" alt="Netflix" />
                    <Header.ButtonLink>Sign In</Header.ButtonLink>
                    <Header.TextLink active={false}>Hello I am a link!</Header.TextLink>
                </Header.Frame>
            </Header>
        );

        expect(getByText('Hello I am a link!')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the full <Header> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <Header src="joker1.jpg" dontShowOnSmallViewPort>
                <Header.Frame>
                    <Header.Group>
                        <Header.Logo src="/images/logo.svg" alt="Netflix" />
                        <Header.TextLink active={false} onClick={() => { }}>
                            Series
            </Header.TextLink>
                        <Header.TextLink active onClick={() => { }}>
                            Films
            </Header.TextLink>
                    </Header.Group>
                    <Header.Group>
                        <Header.Search searchTerm="Joker" setSearchTerm={() => { }} />
                        <Header.Profile>
                            <Header.Picture src="/images/brave.png" />
                            <Header.Dropdown>
                                <Header.Group>
                                    <Header.Picture src="/images/brave.png" />
                                    <Header.TextLink>Brave Leuterio</Header.TextLink>
                                </Header.Group>
                                <Header.Group>
                                    <Header.TextLink onClick={() => { }}>Sign out</Header.TextLink>
                                </Header.Group>
                            </Header.Dropdown>
                        </Header.Profile>
                    </Header.Group>
                </Header.Frame>

                <Header.Feature>
                    <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
                    <Header.Text>Forever alone in a crowd...</Header.Text>
                    <Header.PlayButton>Play</Header.PlayButton>
                </Header.Feature>
            </Header>
        );

        expect(getByTestId('search-input')).toBeTruthy();
        expect(getByTestId('search-input').value).toBe('Joker');
        fireEvent.change(getByTestId('search-input'), { target: { value: 'Simpsons' } });
        fireEvent.click(getByTestId('search-click'));

        expect(getByText('Series')).toBeTruthy();
        expect(getByText('Films')).toBeTruthy();
        expect(getByText('Brave Leuterio')).toBeTruthy();
        expect(getByText('Watch Joker Now')).toBeTruthy();
        expect(getByText('Sign out')).toBeTruthy();
        expect(getByText('Forever alone in a crowd...')).toBeTruthy();
        expect(getByText('Play')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});