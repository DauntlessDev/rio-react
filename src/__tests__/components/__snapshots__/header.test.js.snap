// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background 1`] = `
<div
  class="header__Background-sc-16skmj5-0 usTsY"
  data-testid="header-bg"
>
  <div
    class="header__Darken-sc-16skmj5-2 bjGIMD"
  >
    <div
      class="header__Container-sc-16skmj5-4 gyjpsR"
    >
      <p
        class="header__Link-sc-16skmj5-10 UEgDh"
      >
        Hello I am a link!
      </p>
    </div>
    <div
      class="header__Inner-sc-16skmj5-3 krSShB"
    >
      Inner
    </div>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> without a background 1`] = `
<div
  class="header__Container-sc-16skmj5-4 gyjpsR"
>
  <p
    class="header__Link-sc-16skmj5-10 dRgOtt"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header> with a background 1`] = `
<div
  class="header__Background-sc-16skmj5-0 jcyKvV"
  data-testid="header-bg"
  src="joker1.jpg"
>
  <div
    class="header__Darken-sc-16skmj5-2 bjGIMD"
  >
    <div
      class="header__Container-sc-16skmj5-4 gyjpsR"
    >
      <div
        class="header__Group-sc-16skmj5-1 gbkMa-d"
      >
        <p
          class="header__Link-sc-16skmj5-10 dRgOtt"
        >
          Series
        </p>
        <p
          class="header__Link-sc-16skmj5-10 dRgOtt"
        >
          Films
        </p>
      </div>
      <div
        class="header__Group-sc-16skmj5-1 gbkMa-d"
      >
        <div
          class="header__Search-sc-16skmj5-17 iLDIyI"
        >
          <button
            class="header__SearchIcon-sc-16skmj5-18 ConOt"
            data-testid="search-click"
          >
            <img
              alt="Search"
              src="/images/icons/search.png"
            />
          </button>
          <input
            class="header__SearchInput-sc-16skmj5-16 sGuDS"
            data-testid="search-input"
            placeholder="Search films and series"
            value="Joker"
          />
        </div>
        <div
          class="header__Profile-sc-16skmj5-15 TolnJ"
        >
          <button
            class="header__Picture-sc-16skmj5-13 dvGHv"
            src="/images/users//images/brave.png.png"
          />
          <div
            class="header__Dropdown-sc-16skmj5-14 jZkHRb"
          >
            <div
              class="header__Group-sc-16skmj5-1 gbkMa-d"
            >
              <button
                class="header__Picture-sc-16skmj5-13 dvGHv"
                src="/images/users//images/brave.png.png"
              />
              <p
                class="header__Link-sc-16skmj5-10 dRgOtt"
              >
                Brave Leuterio
              </p>
            </div>
            <div
              class="header__Group-sc-16skmj5-1 gbkMa-d"
            >
              <p
                class="header__Link-sc-16skmj5-10 dRgOtt"
              >
                Sign out
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="header__Container-sc-16skmj5-4 header__Feature-sc-16skmj5-7 gyjpsR cGfnIb"
    >
      <h2
        class="header__FeatureCallOut-sc-16skmj5-11 injLID"
      >
        Watch Joker Now
      </h2>
      <p
        class="header__Text-sc-16skmj5-8 iCXpDq"
      >
        Forever alone in a crowd...
      </p>
      <button
        class="header__PlayButton-sc-16skmj5-19 eofotb"
      >
        Play
      </button>
    </div>
  </div>
</div>
`;
